src/test/core/test_non_wx_functions.cpp

```diff
@@ -0,0 +1,227 @@
+#include "../../core/non_wx_functions.h"
+#include "../../../include/ieee-754-half/half.hpp"
+#include "../../../include/catch2/catch.hpp"
+#include <iostream>
+
+// TODO [Deprecated]: If the forwarded type specific ZeroArray functions do not produce any known errors a year from now, remove the deprecated functions.
+// Now being 2023-Nov-6
+
+TEST_CASE("ZeroBoolArray sets all elements to false", "[ZeroBoolArray]") {
```

Comment:
I suppose I'm a bit confused by what this is; I understand they're all test cases for functions that are not wx-related; but the actual type of each TEST_CASE is not something I've seen before; is it just a type of C/C++ function designed for this type of thing?

Reply:
It is from the [catch2 ](https://github.com/catchorg/Catch2) library. We include their header in include/. TEST_CASE itself is a macro defined by catch2 that does a bunch of neat things under the hood.